mpmp/src/model/Player.java

```diff
@@ -51,8 +63,13 @@ public String toString() {
 	 * Adds or removes money from a player. If the player can't pay, false is returned.
 	 */
 	public boolean addMoney(int sum) {
-		if(money + sum >= 0) {
-			money += sum;
+		if(cash + hyp + sum >= 0) {
+			cash += sum;
+
+			/* cash exhausted; change to hypothecs instead */
+			if(cash < 0)
+				hyp = -cash;
```

Comment:
Don't you need a system to check if you are our of any money?
I don't really get what you are doing here.
If you register a mortgage to your street, you "gain" extra cash money.
Do you see "hyp" as a cache so you then have to think how you get that money?